sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -237,8 +238,12 @@ public void run() {
         if (flushRequested.get() != null) {
           flush();
         }
-        while (!queue.isEmpty() && batch.size() < maxExportBatchSize) {
-          batch.add(queue.poll().toSpanData());
+        if (queue instanceof MessagePassingQueue) {
```

Comment:
Thanks, this optimization makes sense. Can you add it as a static `drainTo` method in `JcTools`? That's where we keep all the code that touches that library and will probably fix the build

Reply:
Thanks @anuraaga for providing feedback, I have updated the changes in a new PR w/o adding more code to JCTool util.

Reply:
@anuraaga created a new PR to address your feedback https://github.com/open-telemetry/opentelemetry-java/pull/4582